File: backend/app/schemas/locations.py
import uuid

from pydantic import BaseModel, Field


class LocationCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    description: str | None = None
    purpose: str | None = None
    goals: str | None = None
    extra_knowledge: str | None = None


class LocationUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    purpose: str | None = None
    goals: str | None = None
    extra_knowledge: str | None = None
    is_active: bool | None = None


class LocationResponse(BaseModel):
    id: uuid.UUID
    business_id: uuid.UUID
    name: str
    slug: str
    description: str | None = None
    purpose: str | None = None
    goals: str | None = None
    extra_knowledge: str | None = None
    is_active: bool

    model_config = {"from_attributes": True}

File: frontend/next/src/dashboard/api/locations.ts
import { api } from "./client";

export interface LocationResponse {
  id: string;
  business_id: string;
  name: string;
  slug: string;
  description: string | null;
  purpose: string | null;
  goals: string | null;
  extra_knowledge: string | null;
  is_active: boolean;
}

export interface LocationCreate {
  name: string;
  description?: string | null;
  purpose?: string | null;
  goals?: string | null;
  extra_knowledge?: string | null;
}

export interface LocationUpdate {
  name?: string;
  description?: string | null;
  purpose?: string | null;
  goals?: string | null;
  extra_knowledge?: string | null;
  is_active?: boolean;
}

export function fetchLocations(token: string) {
  return api.get<LocationResponse[]>("/api/dashboard/locations", token);
}

export function createLocation(token: string, payload: LocationCreate) {
  return api.post<LocationResponse>("/api/dashboard/locations", payload, token);
}

export function updateLocation(token: string, locationId: string, payload: LocationUpdate) {
  return api.patch<LocationResponse>(`/api/dashboard/locations/${locationId}`, payload, token);
}
